### DISCUSSION TITLE
Decrease brightness on SSD1322

### DISCUSSION THREAD

**Discussion by Boro030771** [olikraus/u8g2#1847] [General]
Is there a "Master Current Control" command on SSD1322 in your u8g2 library? 
Sorry for my English.

**Comment by v1993** [comment]
I think what you're looking for is [`sendF`](https://github.com/olikraus/u8g2/wiki/u8g2reference#sendf), which allows using display-specific commands.

**Comment by Boro030771** [comment]
"" Master Current Control (C7h)
This 2 byte command is used to control the output current of the SEG using a scaling factor. The controller has 16 levels of control with coefficients from 1 [0000b] to 16 [1111b, this is the default]. The lower the master current value, the dimmer the OLED panel will be. For example, initially the segment output current is 160 µA with a scaling factor of 16, and setting the scaling factor to 8 will reduce the current to 80 µA.

A[3:0] =
0000b, reduce output current for all colors to 1/16
0001b, output current reduction for all colors to 2/16
..
1110b, output current reduction for all colors to 15/16
1111b, no change (state after reset). ""

  For example - "" u8g2.sendF("C7h", 0000b); "" ?

**Comment by olikraus** [comment]
```
u8g2.sendF("ca", 0xc7, 0x00);
```
For details, see the reference manual, linked by @v1993

**Reply by Boro030771** [reply]
Thank you very much, but for some reason this command does not change the contrast.

**Comment by olikraus** [comment]
What about the setContrast function?

**Reply by Boro030771** [reply]
setContrast(0), this display will be in the car, so the contrast (brightness) is still high at night. 
IREF is the current reference for the ISEG segment current drivers.
ISEG = Contrast / 256 * IREF * scale factor * 2
Here the contrast (0..255) is set by the Set Contrast (C1h) command; scaling factor scale factor (1..16) is set by the Master Current Control (C7h) command. For example, to obtain a current ISEG = 300 µA with a maximum contrast ratio of 255, IREF is set to about 10 µA. The current value is obtained by connecting an external suitable resistor between the IREF pin and the VSS (common) pin, as shown in fig. 8-8. Recommended value IREF = 10 µA.
Remains either Master Current Control or IREF.

**Comment by olikraus** [comment]
Did you try to reduce the pre-charge voltage also?
```
u8g2.sendF("ca", 0xbb, 0x00);
```
I mean, there are many config values which could be modified in order to reduce the brightness.

Another option would be to set a custom gray scale table. U8g2 will only use the code 0x0f, so changing the last value of that table might also help (maybe 0x01 might be too low already):
```
u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01);
```

**Reply by Boro030771** [reply]
Thanks a lot. With u8g2.setContrast(0); , I picked up u8g2.sendF("ca", 0xbb, 0x0A); , turned out great. For smaller values ​​u8g2.sendF("ca", 0xbb, 0x0A); the inscription on the screen does not show stably.

Typed u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01); and now the screen is black. I can't return back. How to return back?

Understood. Dropped the table u8g2.sendF("c", 0xb9);

**Reply by olikraus** [reply]
Instead of 0x01 as last value of the table, use some higher value like 0x07. 0x0f would be the u8g2 default.

**Reply by olikraus** [reply]
I mean
```
u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0f); 
```
should be the default and
```
u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x07); 
```
should have lesser brightness.

**Reply by olikraus** [reply]
I was wrong, the values of the table are between 0 and 180 and must be increasing, so the table should look like this:
Max brightness:
```
u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 180);
```
Half brightness:
```
u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 90);
```
1/4 brightness:
```
u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 45);
```

**Reply by olikraus** [reply]
In principle, you probably need to study the datasheet of the controller. Doing this kind of configuration is very specific to the display controller.

**Reply by Boro030771** [reply]
After u8g2.sendF("caaaaaaaaaaaaaaaa", 0xb8, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 20); , the screen becomes brighter. u8g2.sendF("c", 0xb9); didn't help. After u8g2.sendF("ca", 0xbb, 0x0A); the screen is back to how it was before. I'll leave everything as it was. Thanks again for your help.

**Reply by jareddlc** [reply]
@Boro030771 did you ever figure it out? I have a similar questions here: #2264
